src/utils/apps.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppBinding} from '../types/apps';
+import {AppBindingLocations} from '../constants/apps';
+
+export function fillBindingsInformation(binding?: AppBinding) {
```

Comment:
maybe a better name would be `fixBindings...`/`fillandtrim..`, since we not only populate fields but also trim some.